### PR TITLE
feat: use same permissions used at repo checkout

### LINKED ISSUE #115: Support creating commits via SSH
This would enable the option to use deploy keys.

### PR DESCRIPTION
**Pull request by xsc27**
closes #115

### CODE REVIEW

**Comment by TriPSs** [comment]
@xsc27 thanks for the PR! Could you tell me how does exactly makes it possible to use deploy keys?

**Comment by xsc27** [comment]
I am trying to use the permissions used during checkout, i.e.:

```yaml
- name: Checkout GitHub Action
  uses: actions/checkout@v2
  with:
    ssh-key: ${{ secrets.SSH_DEPLOY_KEY }}
```

so I am expecting that every git operation can continue to use the SSH-key.

I am also expecting that checking out with a token that has write permissions means the token would not have to be passed to `conventional-changelog-action`.

```yaml
- name: Checkout GitHub Action
  uses: actions/checkout@v2
  with:
    token: ${{ secrets.GH_TOKEN }}
```

I hope that makes sense :)

**Comment by TriPSs** [comment]
Okay thanks! Could you also add a bit to the README on how to use the deploy key?

**Comment by xsc27** [comment]
i can. i will update the PR later today :)

File: README.md [hunk 1]
```diff
@@ -188,6 +188,18 @@ Github releases
...
+
+```yaml
+- name: Checkout GitHub Action
+  uses: actions/checkout@v2
+  with:
+    ssh-key: ${{ secrets.SSH_DEPLOY_KEY }}
+- name: Conventional Changelog Action
+  id: changelog
+  uses: TriPSs/conventional-changelog-action@v3
+```
```
Comment (xsc27) [inline README.md:201]: @TriPSs is this adequate?

**Review by TriPSs — APPROVED** [review]

**Pull request merged**